### PR TITLE
PanelEdit: Show when field options have override rules or data config that overrides the default 

### PR DESCRIPTION
**Pull request by torkelo**
This was something that we wanted to do for v7 and v8 but we never had time to complete. 

* [x] Shows a small info dot (orange) when data config is found for an option.  The tooltip currently shows the data fields and config values, not sure if this really needed. Makes the tooltip a bit bigger. 
* [x] Unit test for data config info dots
* [x] Show small info dot (blue) for override rules that override a default  
* [x] Unit test for override rule info dots
* [ ] Find a way to add filter action for each option that has are overridden by an override rule, the filter action would add the option name to option search so you would see the default and all override rules that contain that option and the option values. '

Issues 
* Find our Tooltip component's tooltips a bit hard to read, both the small text & the small padding/contrast/lack of border make them blend into the background too much. 
* Show data override config values or not? For some properties that have object values (data links, color scheme) it will just be json blob which just looks bad. 

Todo
* What colors to use for dots? 
* Feedback from ux team 

Screenshots:
Unit has data override and No value has an override rule that overrides it
Screenshot from 2021-10-10 11-23-25

Options can have both data & override rule overrides
Screenshot from 2021-10-10 11-36-32

data override tooltip, not sure I showing values is needed? For some properties that have object values (data links, color scheme) it will just be json blob which just looks bad. 
Screenshot from 2021-10-10 11-26-10

override rule info dot tooltip:
Screenshot from 2021-10-10 11-27-07

### CODE REVIEW

**Review by hugohaggmark — COMMENTED** [review]
Did a shallow code review and found some small things. 

I think using both orange and blue dot on the same prop looks a bit confusing, maybe we could have one a half orange/half blue dot instead of two?

**Comment by dprokop** [comment]
Found a small issue when one propert has multiple overrides. There are as many dots being rendered as overrides:

image

**Comment by Willena** [comment]
I do like this feature :) Just some naive questions : What would happen if I have 20 (or more) series and override properties gets applied using regex ? Will this result in 20 or more colored dots ? Is the "overflow" of dots handled ?

**Review by dprokop — APPROVED** [review]
Works OK and the code is good too. Can't find anything apart from the yarn pnp things that I think should be checked in in a separate PR.

**Review by hugohaggmark — APPROVED** [review]
Looks good and works as expected 🎉

File: public/app/features/dashboard/components/PanelEditor/OptionsPaneItemOverrides.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+  onClick?: () => void;
+  tooltip: string;
+  description: string;
+}
+
+export const dataOverrideTooltipDescription =
+  'Some data fields have this option pre-configured. Add a field override rule to override the pre-configured value.';
+export const overrideRuleTooltipDescription = 'An override rule exists for this property';
+
+export function searchForOptionOverrides(
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/OptionsPaneItemOverrides.tsx]: :+1:

File: public/app/features/dashboard/components/PanelEditor/OptionsPaneItemOverrides.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+
+        overrideRuleFound = true;
+        break;
+      }
+    }
+
+    if (overrideRuleFound) {
+      break;
+    }
+  }
```
[the remark this answers is not in the document]
Comment (torkelo) [reply public/app/features/dashboard/components/PanelEditor/OptionsPaneItemOverrides.tsx]: great suggestion  !

**Pull request merged**